lib/vue_phoenix/models/user.ex

```diff
@@ -26,27 +26,43 @@ defmodule VuePhoenix.User do
     |> cast(attrs, [:email, :password])
     |> validate_required([:email, :password])
     |> unique_constraint(:email)
+    |> validate_format(:email, ~r/(^[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$)/)
     |> validate_length(:password, min: 6)
     |> validate_confirmation(:password, required: true)
     |> encrypt_password
   end
 
-  def sign_in(email, password) do
-    case Encryption.validate_password(Repo.get_by(User, email: email), password) do
-      {:ok, user} ->
-        token = Authenticator.generate_token(user)
-        Repo.insert(Ecto.build_assoc(user, :tokens, %{code: token}))
-      err -> err
+  def sign_in(%{email: email, password: password}) do
```

Comment:
Not ideal to have `sign_in` or `sign_out` at the model layer. Can use service module for this or apply DDD. 

For example:

```elixir
defmodule Authentication
  def sign_in()
    ...
  end

  def sign_out
    ...
  end
end
```

Reply:
Then can use `Authentication.sign_in ` -> more clean